subsys/partition_manager/Kconfig

```diff
@@ -138,6 +138,20 @@ config PM_EXTERNAL_FLASH_BASE
 
 endif
 
+config BUILD_WITH_TFM
+	bool
+	select PM_SINGLE_IMAGE
+	select ENTROPY_CC3XX
+	select NORDIC_SECURITY_BACKEND
+	imply MBEDTLS_X509_LIBRARY
+	imply MBEDTLS_TLS_LIBRARY
+	imply MBEDTLS_ENABLE_HEAP
+	select NRF_SECURITY_RNG
+	select MBEDTLS_PSA_CRYPTO_C
+	imply MBEDTLS_PSA_CRYPTO_STORAGE_C
+	select MBEDTLS_USE_PSA_CRYPTO
```

Comment:
this looks wrong to me.
There is no safeguard which means that suddenly all those selects and implies becomes active when building with TFM regardless of whether or not the users use nRF Security backend.

Also using both select and imply unconditionally here prevent other part of the tree, like a defconfig to say, as example:
```
config MBEDTLS_X509_LIBRARY
        default n
```
as the imply will always win over a default.

Reply:
Removing x509, tls (trying with CI to see if it is okay)
(This was added as convenience for TLS/DTLS scope examples and was deemed to not have code-impact unless you call the actual APIs)

Reply:
update, I notice there is actually a:
```
select NORDIC_SECURITY_BACKEND
```
in this part.
Is it intended that we enforce the use nRF Security when building with TF-M.
This will effectively block building TF-M for non-nordic boards now, which might be ok.

Reply:
Fixed